File: problem/Matrix_Similarity/areSimilar.c
#include <stdbool.h>

bool areSimilar(int **mat, int matSize, int *matColSize, int k)
{
    int m = matSize;
    int n = matColSize[0];
    k %= n;

    for (int i = 0; i < m; i++)
    {
        for (int j = 0; j < n; j++)
        {
            if (i % 2 == 0)
            {
                if (mat[i][j] != mat[i][(j + k) % n])
                {
                    return false;
                }
            }
            else
            {
                if (mat[i][j] != mat[i][(j - k + n) % n])
                {
                    return false;
                }
            }
        }
    }
    return true;
}

File: problem/Matrix_Similarity/areSimilar.js
/**
 * @param {number[][]} mat
 * @param {number} k
 * @return {boolean}
 */
var areSimilar = function(mat, k) {
    const m = mat.length;
    const n = mat[0].length;
    const shift = k % n;

    for (let i = 0; i < m; i++) {
        for (let j = 0; j < n; j++) {
            if (i % 2 === 0) {
                if (mat[i][j] !== mat[i][(j + shift) % n]) {
                    return false;
                }
            } else {
                if (mat[i][j] !== mat[i][(j - shift + n) % n]) {
                    return false;
                }
            }
        }
    }
    return true;
};